sky/optimizer.py

```diff
@@ -114,11 +124,26 @@ def optimize(dag: 'dag_lib.Dag',
         # node.best_resources if it is None.
         Optimizer._add_dummy_source_sink_nodes(dag)
         try:
-            unused_best_plan = Optimizer._optimize_objective(
-                dag,
-                minimize_cost=minimize == OptimizeTarget.COST,
-                blocked_resources=blocked_resources,
-                quiet=quiet)
+            if _is_dag_resources_ordered(dag):
+                # Honor the user's choice.
+                # Take dag.tasks[0], the actual task
+                # dag can store dummy tasks.
+                resources_list = dag.tasks[0].resources
+                accelerators_str = ', '.join([
+                    r.get_accelerators_str() + r.get_spot_str()
+                    for r in resources_list
+                ])
+                logger.info(
+                    f'{colorama.Fore.YELLOW}Using user-specified accelerators list{colorama.Style.RESET_ALL} (will be tried in the listed order): {accelerators_str}'  # pylint: disable=line-too-long
```

Comment:
It seems we can combine the handling of ordered resources and the normal optimization with the following algorithm (we may not need the _set_resources_by_user_order):

```
def _optimize_dag(dag, task_id):
    local_dag = copy.deepcopy(dag)
    task = dag.tasks[task_id]
    local_task = local_dag.tasks[task_id]
    if isinstance(task.resources, list):
        for resource in task.resources:
            local_task.set_resources(resource)
            if task_id == len(dag.tasks) - 1:
                plan = Optimizer._optimize_objective(local_dag,
                                            minimize_cost=minimize == OptimizeTarget.COST,
                                            blocked_resources=blocked_resources,
                                            quiet = quiet)
                if plan is not None:
                    return plan
            else:
                plan = _optimize_dag(local_dag, task_id + 1)
                if plan is not None:
                    return plan
plan = _optimize_dag(dag, 0)
_set_plan_to_best_resources(dag, plan)
```

Reply:
I think we cannot combine that. Eventually, we want to print out an optimizer table showing different accelerators that the optimizer considered. 
if we do above, it will just show one. We still need to write more code

Reply:
@Michaelvll I think we can leave with `assert len(dag.tasks) == 1, 'Multi-accelerators only Supports one task for the dag'`.
Optimizing multiple tasks for the dag previously require `_optimize_by_dp` or `_optimize_by_ilp` (by cost or time), which are not needed if we just follow user specified order.

Reply:
Addressed. Multi-task dag will print every task's resources (if multiple resources are specified)
also refactored.